### PR TITLE
For styling

### PR DESCRIPTION
**Pull request by bonaxl015**
- Added styles for login, registration, category and task page
- Added some customized methods to get data for progress bar and creator of categories
- Updated tests for login and register

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -3016,7 +3016,15 @@ get-value@^2.0.3, get-value@^2.0.6:
   resolved "https://registry.yarnpkg.com/get-value/-/get-value-2.0.6.tgz#dc15ca1c672387ca76bd37ac0a395ba2042a2c28"
   integrity sha1-3BXKHGcjh8p2vTesCjlbogQqLCg=
 
-glob-parent@^5.1.2, glob-parent@~5.1.2:
+glob-parent@^3.1.0:
```
Comment (Em-Arce) [inline yarn.lock:3268]: cannot bring this back this was deleted due to security
image

File: db/schema.rb [hunk 2]
```diff
@@ -12,6 +11,8 @@
 # It's strongly recommended that you check this file into your version control system.
 
 ActiveRecord::Schema.define(version: 2021_08_07_080258) do
+
```
Comment (Em-Arce) [inline db/schema.rb:14]: not critical

File: app/views/tasks/_index.html.erb [hunk 3]
```diff
@@ -1,31 +1,19 @@
...
-      <th>Priority</th>
-      <th>Deadline</th>
-      <th>Completed</th>
-      <th>Action</th>
-      <th>Action</th>
-    </tr>
-  </thead>
-  <tbody>
+<div class="tasks-container">
+  <div class="d-flex justify-content-evenly flex-wrap">
```
Comment (Em-Arce) [inline app/views/tasks/_index.html.erb:2]: changes due to styling? please confirm

File: app/views/tasks/_form.html.erb [hunk 4]
```diff
@@ -1,42 +1,47 @@
-<%= form_with model: [ @category, @category.tasks.build ] do |f| %>
-  <div>
-    <%= f.label :name %><br>
-    <%= f.text_field :name %><br>
-    <% @category.errors.full_messages_for(:name).each do |message| %>
-      <div class="text-warning"><%= message %></div>
-    <% end %>
+<div class="task-form-container">
+  <div class="title">
+    <h2>Add new task</h2>
```
Comment (Em-Arce) [inline app/views/tasks/_form.html.erb:3]: changes due to styling? please confirm

**Review by Em-Arce — COMMENTED** [review]
Hi have few questions below

File: app/views/tasks/_index.html.erb (hunk 3 above)
Comment (bonaxl015) [reply app/views/tasks/_index.html.erb:2]: Yes

File: app/views/tasks/_form.html.erb (hunk 4 above)
Comment (bonaxl015) [reply app/views/tasks/_form.html.erb:3]: Yes

File: yarn.lock [hunk 5]
```diff
@@ -9,38 +9,38 @@
   dependencies:
     "@babel/highlight" "^7.14.5"
 
-"@babel/compat-data@^7.13.11", "@babel/compat-data@^7.14.5", "@babel/compat-data@^7.14.7":
-  version "7.14.7"
-  resolved "https://registry.yarnpkg.com/@babel/compat-data/-/compat-data-7.14.7.tgz#7b047d7a3a89a67d2258dc61f604f098f1bc7e08"
-  integrity sha512-nS6dZaISCXJ3+518CWiBfEr//gHyMO02uDxBkXTKZDN5POruCnOZ1N4YBRZDCabwF8nZMWBpRxIicmXtBs+fvw==
+"@babel/compat-data@^7.13.11", "@babel/compat-data@^7.14.7", "@babel/compat-data@^7.15.0":
```
Comment (Em-Arce) [inline yarn.lock:12]: please undo this commit, cannot review yet as I have tests to do

**Review by Em-Arce — COMMENTED** [review]
Re- conflicts: since the changes are for styling and does not affect logic, please resolve conflicts: 

do the git merge process and edit files by adding your styles then push.

**Comment by bonaxl015** [comment]
Issues already resolved.

**Pull request merged**